flexget/plugins/list/entry_list.py

```diff
@@ -0,0 +1,109 @@
+from __future__ import unicode_literals, division, absolute_import
+
+import logging
+from collections import MutableSet
+from datetime import datetime
+
+from sqlalchemy import Column, Unicode, PickleType, Integer, DateTime, or_
+from sqlalchemy.sql.elements import and_
+
+from flexget import plugin
+from flexget.db_schema import versioned_base
+from flexget.entry import Entry
+from flexget.event import event
+from flexget.utils.database import safe_pickle_synonym, with_session
+
+log = logging.getLogger('entry_list')
+Base = versioned_base('entry_list', 0)
+
+
+class StoredEntry(Base):
+    __tablename__ = 'entry_list'
+    id = Column(Integer, primary_key=True)
+    list = Column(Unicode, index=True)
+    added = Column(DateTime, default=datetime.now)
+    title = Column(Unicode)
+    original_url = Column(Unicode)
+    _entry = Column('entry', PickleType)
+    entry = safe_pickle_synonym('_entry')
+
+    def __init__(self, list, entry):
+        self.list = list
+        self.title = entry['title']
+        self.original_url = entry['original_url']
+        self.entry = entry
+
+    def __repr__(self):
+        return '<StoredEntry,title=%s,original_url=%s>' % (self.title, self.original_url)
+
+
+class DBEntrySet(MutableSet):
+    def __init__(self, config):
+        self.config = config
+
+    def _query(self, session):
+        return session.query(StoredEntry).filter(StoredEntry.list == self.config)
+
+    def _entry_query(self, session, entry):
+        query = self._query(session)
+        db_entry = query.filter(or_(
+            StoredEntry.title == entry['title'], and_(
+                StoredEntry.original_url,
+                StoredEntry.original_url == entry['original_url']))).first()
+        if db_entry:
+            return db_entry
+
+    @with_session
+    def __iter__(self, session=None):
+        return (Entry(e.entry) for e in self._query(session).order_by(StoredEntry.added.desc()).all())
```

Comment:
@gazpachoking unlike movie list, entry_list iterate over its inner entry attribute and does not return the whole object. This means that by using that standard set call I cannot access that data from the API/CLI. I'm not sure how to proceed, as it would required either changing the __iter__ method or adding a method that will fetch the data from the DB by will required me to override the set built-in methods (at least for the get all API operation).